### PR TITLE
TCG's Mass Client Push 2.0

### PR DESCRIPTION
**Pull request by OfficialTCGMatt**
Moved Error Handler, converted $.get and $.post to $.ajax (excluding one) to allow good error checking, cut a lot of uneeded lines while checking for ajax status errors, ate a few bytes.

### CODE REVIEW

File: functions.js [hunk 1]
```diff
@@ -24,6 +77,7 @@ let MakeWindowsInteractive = function () {
   document.querySelectorAll("window:not([noresize])").forEach((win) => {
     let title = win.querySelector(".window-title");
     title.ondblclick = function () {
+      if (meowos.error) return;
```
Comment (itzTheMeow) [inline functions.js:80]: if theres an error, this will never be called, as the page will be an orange-screen

File: functions.js [hunk 2]
```diff
@@ -37,6 +91,7 @@ let MakeWindowsInteractive = function () {
   // Init minimize buttons.
   document.querySelectorAll(".window-minimize").forEach((minimize) => {
     minimize.onclick = function () {
+      if (meowos.error) return;
```
Comment (itzTheMeow) [inline functions.js:94]: if theres an error, this will never be called, as the page will be an orange-screen

File: functions.js [hunk 3]
```diff
@@ -200,18 +258,24 @@ let UninstallProgram = function (prog) {
...
     },
-    (err) => {
-      if (err.error) console.error(err); //TODO: add orangescreen
+    success: function (data, text) {
       window.location.reload();
+    },
+    error: function (request, status, error) {
+      if (meowos.error) return;
+
+      meowos.ErrorHandler("SYSTEM", `Failed to uninstall ${prog} due to: ${AjaxErrorMsg(request, status, error)}`, true);
```
Comment (itzTheMeow) [inline functions.js:276]: it would be "failed to save file", as its a file?

File: functions.js [hunk 4]
```diff
@@ -495,3 +579,71 @@ _("overlay-create-login").onclick = function () {
...
+
+meowos.ErrorHandler = function (
+  location = "SYSTEM",
+  error,
+  reportToDevs = false
+) {
+  var errorData = error.message ? error.message : error;
+  if (error.stack) errorData += `\nStacktrace: ${error.stack}`;
+  document.body.innerHTML =
+    "<style>" +
```
Comment (itzTheMeow) [inline functions.js:597]: maybe make this a template string? (with the `)

File: script.js [hunk 5]
```diff
@@ -150,6 +61,8 @@ let ONLOAD = function () {
   setInterval(SaveWindows, 3000); //TODO: make this save when windows are moved (prevents API spam)
 
   _("taskbar-button").onclick = function () {
+    if (meowos.error) return;
```
Comment (itzTheMeow) [inline script.js:64]: if theres an error, this will never be called, as the page will be an orange-screen

**Review by itzTheMeow — CHANGES_REQUESTED** [review]

File: functions.js (hunk 2 above)
Comment (OfficialTCGMatt) [reply functions.js:94]: yeah, but it spams the console for some reason, soooo

File: functions.js (hunk 1 above)
Comment (OfficialTCGMatt) [reply functions.js:80]: yeah, but it spams the console for some reason, soooo

File: functions.js (hunk 3 above)
Comment (OfficialTCGMatt) [reply functions.js:276]: I guess so, gonna do that now.

File: functions.js (hunk 4 above)
Comment (OfficialTCGMatt) [reply functions.js:597]: k

**Review by itzTheMeow — APPROVED** [review]

**Pull request merged**